### PR TITLE
Offline detection UI

### LINKED ISSUE #2520: Create an offline detection mechanism(GSoC)
**Is your feature request related to a problem? Please describe.**
Currently, Talawa lacks an offline detection mechanism. 

Like other social media apps, the application should support the following features:
1. Detect the mobile device network status.
2. Gracefully prompt the user about his network status.

**Describe the solution you'd like**
1. Develop a `ConnectivityService` to keep track of device network status.
2. Create UI components similar to `YouTube` to prompt the user about network status.

offline
online

**Describe alternatives you've considered**

**Approach to be followed (optional)**
A clear and concise description of approach to be followed.

**Additional context**

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Azad99-9**
**What kind of change does this PR introduce?**

This PR integrates the [offline detection business logic](https://github.com/PalisadoesFoundation/talawa/pull/2523) into UI.

**Issue Number:**

Fixes #2520 

**Did you add tests for your changes?**

Yes added tests.

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa/assets/120930148/ad3a3c7d-8f33-48d7-80a3-ae700cc67f7a

**If relevant, did you update the documentation?**

**Summary**

The motivation is to give the end-user a responsive UI feed back about his network connectivity status.

**Does this PR introduce a breaking change?**

No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
@noman2002 PTAL

**Comment by Azad99-9** [comment]
@noman2002 please hold this PR don't merge. A bit of patch work is left for this. I will promptly intimate when this PR is ready to merge.

**Comment by Azad99-9** [comment]
@noman2002 This PR is ready for merging.

**Comment by Azad99-9** [comment]
@noman2002 This PR is ready to merge.

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -17,6 +17,7 @@
         android:label="talawa"
         android:icon="@mipmap/ic_launcher"
         android:allowBackup="false"
+        android:enableOnBackInvokedCallback="true"
```
Comment (Azad99-9) [inline android/app/src/main/AndroidManifest.xml:20]: Edited for custom back navigation.

**Review by noman2002 — APPROVED** [review]
LGTM

**Review by palisadoes — APPROVED** [review]

**Pull request merged**